frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -384,6 +435,28 @@ class TreeHierarchyView extends React.PureComponent<Props, State> {
         >
           <i className="fas fa-eye" /> Hide/Show all other trees
         </Menu.Item>
+        <Menu.Item
+          key="shuffleTreeGroupColors"
+          onClick={() => {
+            if (id === MISSING_GROUP_ID) this.props.onShuffleAllTreeColors();
+            else this.shuffleTreeGroupColors(id);
+          }}
+          title="Shuffle Tree Colors"
+        >
+          <i className="fas fa-adjust" /> Shuffle Tree Group Colors
+        </Menu.Item>
+        {id !== MISSING_GROUP_ID ? (
```

Comment:
The group to tree accessors/helpers do not work with the root group. Hence, we can not offer this option on the root group.

Reply:
For the sake of completeness/consistency, I'd add an extra method for this case (similar to the shuffle-all-trees case). I think, it should quite short. Something like:

```js
function ... {
    const setTreeColorActions = Object.keys(this.props.trees).map((treeId) => setTreeColorAction(treeId, color));
    this.props.onBatchActions(setTreeColorActions, "SET_TREE_COLOR");
}
```

Reply:
Good idea. I added such a method.